core/src/main/java/org/mule/runtime/core/DefaultMuleMessage.java

```diff
@@ -777,13 +771,17 @@ public String getCorrelationId()
     public void setReplyTo(Object replyTo)
     {
         assertAccess(WRITE);
+        this.replyTo = replyTo;
         if (replyTo != null)
         {
-            setProperty(MuleProperties.MULE_REPLY_TO_PROPERTY, replyTo, PropertyScope.OUTBOUND);
+            if(replyTo instanceof Serializable)
+            {
+                setProperty(MuleProperties.MULE_REPLY_TO_PROPERTY, (Serializable) replyTo, OUTBOUND, null);
```

Comment:
This may be removed in future PR's but for now want to maintain as much compatibility as possible.